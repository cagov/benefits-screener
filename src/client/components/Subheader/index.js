import BPOButton from "../BPOButton";
import Dropdown from "react-bootstrap/Dropdown";
import React from "react";
import i18n from "../../i18n";
import { useTranslation } from "react-i18next";

function Subheader() {
  const { t } = useTranslation();

  const changeLanguage = (lng) => {
    i18n.changeLanguage(lng);
  };

  return (
    <div className="bg-light">
      <div className="container p-4">
        <h1>{t("subheaderHeader")}</h1>
        <p>{t("subheaderSubheader")}</p>
        <p>{t("subheaderParagraph")}</p>
        <p>{t("subheaderParagraph2")}</p>
        <div className="btn-toolbar" role="toolbar">
          <div className="mr-4 mb-4">
            <BPOButton />
          </div>
          <Dropdown>
            <Dropdown.Toggle
              variant="outline-secondary"
              className="text-dark bg-light"
            >
              Select Language
            </Dropdown.Toggle>

            <Dropdown.Menu>
              <Dropdown.Item onClick={() => changeLanguage("en")}>
                English
              </Dropdown.Item>
              <Dropdown.Item onClick={() => changeLanguage("es")}>
                EspaÃ±ol
              </Dropdown.Item>
            </Dropdown.Menu>
          </Dropdown>
        </div>
      </div>
    </div>
  );
}

export default Subheader;
