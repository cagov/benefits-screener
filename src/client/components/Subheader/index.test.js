import React from "react";
import Subheader from "./index";
import { shallow } from "enzyme";

describe("<Subheader />", () => {
  it("renders vertical navigation bar", async () => {
    const wrapper = shallow(<Subheader />);

    expect(wrapper).toMatchSnapshot();
  });
});
