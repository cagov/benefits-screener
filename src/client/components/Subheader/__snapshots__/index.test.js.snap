// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subheader /> renders vertical navigation bar 1`] = `
<div
  className="bg-light"
>
  <div
    className="container p-4"
  >
    <h1>
      Guide to applying for unemployment benefits
    </h1>
    <p>
      Learn what type of benefits you qualify for and how to apply for them.
    </p>
    <p>
      If you are out of work or had your hours reduced, you may be eligible to receive unemployment benefits from California’s Employment Development Department (EDD).
    </p>
    <p>
      First register or log in at Benefit Programs Online, then apply for unemployment benefits on UI Online℠.
    </p>
    <div
      className="btn-toolbar"
      role="toolbar"
    >
      <div
        className="mr-4 mb-4"
      >
        <BPOButton />
      </div>
      <Dropdown
        navbar={false}
      >
        <DropdownToggle
          className="text-dark"
          variant="outline-secondary"
        >
          Select Language
        </DropdownToggle>
        <DropdownMenu
          alignRight={false}
          flip={true}
        >
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            English
          </DropdownItem>
          <DropdownItem
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            onClick={[Function]}
          >
            Español
          </DropdownItem>
        </DropdownMenu>
      </Dropdown>
    </div>
  </div>
</div>
`;
